sdks/python/apache_beam/runners/common.py

```diff
@@ -39,6 +39,75 @@
 from apache_beam.utils.windowed_value import WindowedValue
 
 
+class NameContext(object):
+  """Holds the name information for a step."""
+
+  def __init__(self, step_name):
+    """Creates a new step NameContext.
+
+    Args:
+      step_name: The name of the step.
+    """
+    self.step_name = step_name
+
+  def __eq__(self, other):
+    return self.step_name == other.step_name
+
+  def __ne__(self, other):
+    return not self == other
+
+  def __repr__(self):
+    return 'NameContext(%s)' % self.__dict__()
+
+  def __hash__(self):
+    return hash(self.step_name)
+
+  def metrics_name(self):
+    """Returns the step name used for metrics reporting."""
+    return self.step_name
+
+  def logging_name(self):
+    """Returns the step name used for logging."""
+    return self.step_name
+
+
+class DataflowNameContext(NameContext):
```

Comment:
<!--new_thread; commit:9af473735321a610ce116d3a9dcf5486a68db57e; resolved:0-->
Do we need this to be exposed in the Beam code?  Can we have this only in the worker?  If this needs to be done in two steps, can you add a TODO / Jira?

Reply:
For now, yes. This is because the Dataflow-specific setup is done in `create_operation`. This will be followed by 1) internal change, and then 2)PR removing it.